openadapt/config.py

```diff
@@ -132,35 +134,57 @@ def obfuscate(val, pct_reveal=0.1, char="*"):
     return rval
 
 
-
 _OBFUSCATE_KEY_PARTS = ("KEY", "PASSWORD", "TOKEN")
 if multiprocessing.current_process().name == "MainProcess":
     for key, val in dict(locals()).items():
         if not key.startswith("_") and key.isupper():
             parts = key.split("_")
             if (
-                any([part in parts for part in _OBFUSCATE_KEY_PARTS]) and
-                val != _DEFAULTS[key]
+                any([part in parts for part in _OBFUSCATE_KEY_PARTS])
+                and val != _DEFAULTS[key]
             ):
                 val = obfuscate(val)
             logger.info(f"{key}={val}")
 
 
-def filter_log_messages(data):
-    """
-    This function filters log messages by ignoring any message that contains a specific string.
+def filter_log_messages(
+    data: logger,
+    max_num_warnings_per_second: int = MAX_NUM_WARNINGS_PER_SECOND,
+) -> bool:
+    """This function filters log messages by ignoring any message that contains a specific string,
+      up to a maximum number of warnings per second.
 
     Args:
-      data: The input parameter "data" is expected to be data from a loguru logger.
+        data: The input parameter "data" is expected to be data from a loguru logger.
+        max_num_warnings_per_second: The maximum number of warnings allowed per second.
+          Set to 0 to disable.
 
     Returns:
-      a boolean value indicating whether the message in the input data should be ignored or not. If the
-    message contains any of the messages in the `messages_to_ignore` list, the function returns `False`
-    indicating that the message should be ignored. Otherwise, it returns `True` indicating that the
-    message should not be ignored.
+        A boolean value indicating whether the message in the input data should be ignored or not.
+        If the maximum number of warnings per second is exceeded, the function returns False,
+        indicating that the message should be ignored. Otherwise, it returns True,
+        indicating that the message should not be ignored.
     """
-    # TODO: ultimately, we want to fix the underlying issues, but for now, we can ignore these messages
     messages_to_ignore = [
         "Cannot pickle Objective-C objects",
     ]
-    return not any(msg in data["message"] for msg in messages_to_ignore)
+
+    # Track the timestamps of each message
+    message_timestamps = defaultdict(list)
+
+    for msg in messages_to_ignore:
+        if msg in data["message"]:
+            if max_num_warnings_per_second > 0:
+                current_timestamp = time.time()
+                message_timestamps[msg].append(current_timestamp)
+                timestamps = message_timestamps[msg]
+
+                # Remove timestamps older than 1 second
+                timestamps = [ts for ts in timestamps if current_timestamp - ts <= 1]
```

Comment:
What about making `1` e.g. `config.WARNING_SUPPRESSION_PERIOD` or similar

Reply:
resolved in https://github.com/OpenAdaptAI/OpenAdapt/pull/389/commits/29ac2e0f5714c4c1ae46946ee84f179d3d73b6b9